Bug: Multi-return type inference fails for direct stdlib calls via 'using', causing false type mismatch errors

## Description

When calling stdlib functions directly (without module prefix) via `import & use`, multi-return type inference fails. Variables from multi-return assignments are registered with empty types, causing false type mismatch errors when those variables are used in return statements.

## Reproduction

```ez
module utils

import & use @std, @io

do check_if_file_is_empty(file string) -> (bool, Error){
    temp data, readErr = read_bytes(file)  // Called directly via 'using'
    if readErr != nil {
        return false, readErr  // ERROR: type mismatch
    } or len(data) > 0 {
        return false, nil
    }
    return true, nil
}
```

## Error Message

```
error[E3012]: return type mismatch: expected Error, got
  --> utils/helpers.ez:80:3
   |
80 |         return false, readErr
   |   ^ return type does not match declaration
```

Note: "got" shows nothing because `readErr` has an empty type.

## Root Cause

In `pkg/typechecker/typechecker.go`, `checkMultiReturnDeclaration()`:

1. **Lines 2051-2066**: When checking builtin multi-return types, `getBuiltinMultiReturnTypes(funcName)` is called
2. **Lines 4882-4889**: `getBuiltinMultiReturnTypes()` only handles `read_int`, not `read_bytes` or other stdlib functions
3. **Lines 2068-2075**: Since the function isn't found, variables are registered with empty type `""`
4. Return statement validation then compares `""` vs `"Error"` which fails

The issue is that `getModuleMultiReturnTypes()` (lines 4891-5034) correctly defines all io functions like `read_bytes` returning `[]string{"[byte]", "error"}`, but this is only checked for module-prefixed calls (e.g., `io.read_bytes()`), not for direct calls via `using`.

## Suggested Fix

In `checkMultiReturnDeclaration()`, after the `getBuiltinMultiReturnTypes()` check fails, also check `getModuleMultiReturnTypes()` for modules that are imported with `using`:

```go
if builtinReturnTypes == nil {
    // Check if it's a stdlib function from a 'using' module
    for moduleName := range tc.fileUsingModules {
        if moduleReturnTypes := tc.getModuleMultiReturnTypes(moduleName, funcName, args); moduleReturnTypes != nil {
            for i, name := range decl.Names {
                if name != nil {
                    inferredType := ""
                    if i < len(moduleReturnTypes) {
                        inferredType = moduleReturnTypes[i]
                    }
                    tc.defineVariableWithMutability(name.Value, inferredType, decl.Mutable)
                }
            }
            return
        }
    }
}
```

## Related Issues

- #909 - `using` directive does not bring stdlib types, constants, and exports into scope (related `using` issue)
- #944 - Similar symptom (empty type) but different cause (single-variable assignment)

## Environment

- EZ version: latest main branch
- Affected functions: All multi-return stdlib functions when called via `using` (io.read_bytes, io.read_file, io.read_dir, os.get_env, json.decode, etc.)

--- a/pkg/typechecker/typechecker.go
+++ b/pkg/typechecker/typechecker.go
@@ -2238,6 +2238,23 @@ func (tc *TypeChecker) checkMultiReturnDeclaration(decl *ast.VariableDeclaration
 			return
 		}
 
+		// Check if it's a stdlib function from a 'using' module (#977)
+		for usingModuleName := range tc.fileUsingModules {
+			resolvedModuleName := tc.resolveStdlibModule(usingModuleName)
+			if moduleReturnTypes := tc.getModuleMultiReturnTypes(resolvedModuleName, funcName, callExpr.Arguments); moduleReturnTypes != nil {
+				for i, name := range decl.Names {
+					if name != nil {
+						inferredType := ""
+						if i < len(moduleReturnTypes) {
+							inferredType = moduleReturnTypes[i]
+						}
+						tc.defineVariableWithMutability(name.Value, inferredType, decl.Mutable)
+					}
+				}
+				return
+			}
+		}
+
 		// Function not found - still register variables with unknown types
 		// (undefined function error will be caught elsewhere)
 		for _, name := range decl.Names {

